ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala

```diff
@@ -262,6 +275,282 @@ final class CommandService(session: LedgerSession) extends LedgerTestSuite(sessi
+      context =>
+        {
+          val target = 15
+          for {
+            ledger <- context.participant()
+            party <- ledger.allocateParty()
+            commands = Vector.fill(target)(Dummy(party).create.command)
+            request <- ledger.submitAndWaitRequest(party, commands: _*)
+            _ <- ledger.submitAndWait(request)
+            acs <- ledger.activeContracts(party)
+          } yield {
+            assert(
+              acs.size == target,
+              s"Expected $target contracts to be created, got ${acs.size} instead")
+          }
+        }
+    }
+
+  private[this] val callablePayout =
+    LedgerTest("CSCallablePayout", "Run CallablePayout and return the right events") { context =>
+      for {
+        Vector(alpha, beta) <- context.participants(2)
+        Vector(giver, newReceiver) <- alpha.allocateParties(2)
+        receiver <- beta.allocateParty()
+        callablePayout <- alpha.create(giver, CallablePayout(giver, receiver))
+        tree <- beta.exercise(receiver, callablePayout.exerciseTransfer(_, newReceiver))
+      } yield {
+        val created = assertSingleton("There should only be one creation", createdEvents(tree))
+        assertEquals(
+          "The created event should be the expected one",
+          created.getCreateArguments.fields,
+          encode(CallablePayout(giver, newReceiver)).getRecord.fields)
+      }
+    }
+
+  private[this] val unitAsArgumentToNothing =
+    LedgerTest("CSUnitAsArgumentToNothing", "Return Unit as argument to Nothing") { context =>
+      for {
+        ledger <- context.participant()
+        party <- ledger.allocateParty()
+        template = NothingArgument(party, Primitive.Optional.empty)
+        _ <- ledger.create(party, template)
+        acs <- ledger.activeContracts(party)
+      } yield {
+        val contract = assertSingleton("More than one active contract", acs)
+        assertEquals(
+          "Contract mismatch",
+          contract.getCreateArguments.fields,
+          encode(template).getRecord.fields)
+      }
+    }
+
+  private[this] val readyForExercise =
+    LedgerTest(
+      "CSReadyForExercise",
+      "It should be possible to exercise a choice on a created contract") { context =>
+      for {
+        ledger <- context.participant()
+        party <- ledger.allocateParty()
+        factory <- ledger.create(party, DummyFactory(party))
+        tree <- ledger.exercise(party, factory.exerciseDummyFactoryCall)
+      } yield {
+        val exercise = assertSingleton("There should only be one exercise", exercisedEvents(tree))
+        assert(exercise.contractId == factory.unwrap, "Contract identifier mismatch")
+        assert(exercise.consuming, "The choice should have been consuming")
+        val _ = assertLength("Two creations should have occured", 2, createdEvents(tree))
+      }
+    }
+
+  private[this] val completions =
```

Comment:
Very strictly speaking this test doesn't test the `CommandService` but the `CommandCompletionService`. Or did you plan to put all the tests that are command related in this file?

Reply:
I'd keep it here for now and move it when we actually port the `CommandCompletionServiceIT` suite. Sounds good?

Reply:
Sure.